plugins/techdocs-backend/src/techdocs/stages/prepare/dir.ts

```diff
@@ -13,23 +13,80 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import fs from 'fs-extra';
+import os from 'os';
 import { PreparerBase } from './types';
 import { Entity } from '@backstage/catalog-model';
 import path from 'path';
 import { parseReferenceAnnotation } from './helpers';
+import { InputError } from '@backstage/backend-common';
+import { Clone } from 'nodegit';
+import GitUriParser from 'git-url-parse';
 
 export class DirectoryPreparer implements PreparerBase {
-  prepare(entity: Entity): Promise<string> {
-    const { location: managedByLocation } = parseReferenceAnnotation(
+  private async cloneGithubRepo(entity: Entity) {
+    const { protocol, location } = parseReferenceAnnotation(
```

Comment:
they are referred to as `type` and `target` in the code actually (because they aren't always protocols or locations as such).